pkg/upgrade/upgrade.go

```diff
@@ -72,6 +72,15 @@ func ManagedInstances(ctx context.Context, c client.Client, reader client.Reader
 					"namespace": jaeger.Namespace,
 				}).WithError(err).Error("failed to store the upgraded instance")
 				tracing.HandleError(err, span)
+			} else {
```

Comment:
It's more readable if you invert the situation: "if err == nil { success } else { failure }". We use "if err != nil" only when that's the only code branch.

Also, you might want to change from Update to Patch.

Reply:
ok, that make sense, about changing Update to Patch I would prefer to do it in a separate PR if it's required.